include/lp_hash_table.h

```diff
@@ -0,0 +1,93 @@
+#ifndef DIL_LP_HASH_TABLE_H
+#define DIL_LP_HASH_TABLE_H
+
+#include <assert.h>
+#include <functional>
+#include <stdint.h>
+#include <stdlib.h>
+#include <type_traits>
+
+template <typename T> struct lp_hash_table {
```

Comment:
People choose a way to make clear that something is a struct by the naming.
That's usually either by putting a `_t` or using some different naming convention.
I will upload a coding guide soon and I propose that we use `snake_case` generally, but for structs `PascalCase`.
It seems to be the most effective convention I have worked with. So you may rename that to e.g. `IdHashTable` or something like that. It will probably change anyway when we find uses for it.